syntax = "proto3";

package order;

service OrderService {
  rpc CreateOrder (CreateOrderRequest) returns (CreateOrderResponse);
  rpc GetOrder (GetOrderRequest) returns (GetOrderResponse);
  rpc UpdateOrderStatus (UpdateOrderStatusRequest) returns (UpdateOrderStatusResponse);
  rpc ListUserOrders (ListUserOrdersRequest) returns (ListUserOrdersResponse);
}

message CreateOrderRequest {
  string user_id = 1;
  string restaurant_id = 2;
  string items = 3;  // JSON string of items
  double total_amount = 4;
  string delivery_address = 5;
  string payment_method = 6;
}

message CreateOrderResponse {
  string order_id = 1;
  string status = 2;
  string message = 3;
}

message GetOrderRequest {
  string order_id = 1;
}

message GetOrderResponse {
  string order_id = 1;
  string user_id = 2;
  string restaurant_id = 3;
  string items = 4;
  string status = 5;
  double total_amount = 6;
  string delivery_address = 7;
  string created_at = 8;
}

message UpdateOrderStatusRequest {
  string order_id = 1;
  string status = 2;
}

message UpdateOrderStatusResponse {
  string order_id = 1;
  string status = 2;
  string message = 3;
}

message ListUserOrdersRequest {
  string user_id = 1;
  int32 limit = 2;
  int32 offset = 3;
}

message ListUserOrdersResponse {
  repeated GetOrderResponse orders = 1;
  int32 total = 2;
}
